core/src/autogluon/core/hpo/executors.py

```diff
@@ -60,17 +61,29 @@ def initialize(
         """
         raise NotImplementedError
     
-    @abstractmethod
-    def register_resources(self, resources: dict):
+    def register_resources(self, initialized_model: "AbstractModel"):
```

Comment:
I suggest using [PEP 563: Postponed Evaluation of Annotations](https://docs.python.org/3.7/whatsnew/3.7.html#pep-563-postponed-evaluation-of-annotations) instead of string.